src/pocapi.cpp

```diff
@@ -80,3 +94,145 @@ OVMS_Status* OVMS_ServerMultiModelOptionsSetConfigPath(OVMS_ServerMultiModelOpti
+    // TODO error handling if null
+    // if (nullptr == req)
+    InferenceRequest* request = reinterpret_cast<InferenceRequest*>(req);
+    auto status = request->setInputBuffer(inputName, data, bufferSize, bufferType, deviceId);
+    if (!status.ok()) {
+        return reinterpret_cast<OVMS_Status*>(new Status(status));
+    }
+    return nullptr;
+}
+
+OVMS_Status* OVMS_InferenceRequestAddParameter(OVMS_InferenceRequest* req, const char* parameterName, OVMS_DataType datatype, const void* data, size_t byteSize) {
+    // TODO error handling if null
+    // if (nullptr == req)
+    InferenceRequest* request = reinterpret_cast<InferenceRequest*>(req);
+    auto status = request->addParameter(parameterName, datatype, data);
+    if (!status.ok()) {
+        return reinterpret_cast<OVMS_Status*>(new Status(status));
+    }
+    return nullptr;
+}
+
+OVMS_Status* OVMS_InferenceRequestRemoveParameter(OVMS_InferenceRequest* req, const char* parameterName) {
+    // TODO error handling if null
+    // if (nullptr == req)
+    InferenceRequest* request = reinterpret_cast<InferenceRequest*>(req);
+    auto status = request->removeParameter(parameterName);
+    if (!status.ok()) {
+        return reinterpret_cast<OVMS_Status*>(new Status(status));
+    }
+    return nullptr;
+}
+
+OVMS_Status* OVMS_InferenceRequestRemoveInput(OVMS_InferenceRequest* req, const char* inputName) {
+    // TODO error handling if null
+    // if (nullptr == req)
+    InferenceRequest* request = reinterpret_cast<InferenceRequest*>(req);
+    auto status = request->removeInput(inputName);
+    if (!status.ok()) {
+        return reinterpret_cast<OVMS_Status*>(new Status(status));
+    }
+    return nullptr;
+}
+
+OVMS_Status* OVMS_InferenceRequestInputRemoveData(OVMS_InferenceRequest* req, const char* inputName) {
+    // TODO error handling if null
+    // if (nullptr == req)
+    InferenceRequest* request = reinterpret_cast<InferenceRequest*>(req);
+    auto status = request->removeInputBuffer(inputName);
+    if (!status.ok()) {
+        return reinterpret_cast<OVMS_Status*>(new Status(status));
+    }
+    return nullptr;
+}
+
+OVMS_Status* OVMS_InferenceResponseGetOutput(OVMS_InferenceResponse* res, uint32_t id, const char** name, OVMS_DataType* datatype, const uint64_t** shape, uint32_t* dimCount, void** data, size_t* bytesize, BufferType* bufferType, uint32_t* deviceId) {
+    // TODO error handling if null
+    // if (nullptr == req)
+    InferenceResponse* response = reinterpret_cast<InferenceResponse*>(res);
+    InferenceTensor* tensor = nullptr;
+    const std::string* cppName;
+    auto status = response->getOutput(id, &cppName, &tensor);
+    if (!status.ok() ||
+        (tensor == nullptr) ||
+        (cppName == nullptr)) {
+        return reinterpret_cast<OVMS_Status*>(new Status(status));
```

Comment:
It is possible to return status ok here. (if cppName == nullptr or tensor == nullptr)

Reply:
getOutput is tested separately -> if you get OK then you have to have both tensor & name

Reply:
It's rather a matter of why block API user from reusing that buffer after completion of inference & reading outputs & deleteing the response? For now you have to hold with the deletion of InferenceResponse.

Reply:
What I mean was that it is possible to exit the function with non nullptr value, to status code which holds ovms::StatusCode::OK which is not desired.

It will happen when status is OK in line 181 but tensor is nullptr or cppName is nullptr.